storm-client/src/jvm/org/apache/storm/messaging/netty/Client.java

```diff
@@ -163,6 +170,50 @@
             waitStrategy = ReflectionUtils.newInstance(clazz);
         }
         waitStrategy.prepare(topoConf, WaitSituation.BACK_PRESSURE_WAIT);
+        this.metricRegistry = metricRegistry;
+
+        // it's possible to be passed a null metric registry if users are using their own IContext implementation.
+        if (this.metricRegistry != null) {
+            Gauge<Integer> reconnects = new Gauge<Integer>() {
+                @Override
+                public Integer getValue() {
+                    return totalConnectionAttempts.get();
+                }
+            };
+            metricRegistry.gauge("__send-connection-reconnects-" + host + ":" + port, reconnects,
```

Comment:
We are using different names here and in the doc docs/Metrics.md

`__send-iconnection` vs `__send-connection`

I would prefer to preserve the original name since we have been using it in previous storm versions. But at least let's change one of them and make it consistent.